processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -367,6 +368,10 @@ public boolean isArrayType() {
         return componentType != null;
     }
 
+    public boolean isOptionalType() {
+        return Optional.class.getName().equals( getFullyQualifiedName() );
+    }
```

Comment:
This is only used in the `getNull`, so it shouldn't be `public`. Additionally, we should also support `OptionalInt`, `OptionalDouble` and `OptionalLong`.

Reply:
done